Code/GraphMol/MolInterchange/Parser.cpp

```diff
@@ -205,6 +207,143 @@ void readBond(RWMol *mol, const rj::Value &bondVal,
   }
 }
 
+template <typename T>
+void parseProperties(T &obj, const rj::Value &propsVal) {
+  for (const auto &propVal : propsVal.GetObject()) {
+    if (propVal.value.IsInt()) {
+      obj.setProp(propVal.name.GetString(), propVal.value.GetInt());
+    } else if (propVal.value.IsDouble()) {
+      obj.setProp(propVal.name.GetString(), propVal.value.GetDouble());
+    } else if (propVal.value.IsString()) {
+      obj.setProp(propVal.name.GetString(), propVal.value.GetString());
+    }
+  }
+}
+
+void readStereoGroups(RWMol *mol, const rj::Value &sgVals) {
+  PRECONDITION(mol, "no mol");
+
+  std::vector<StereoGroup> molSGs(mol->getStereoGroups());
+  for (const auto &sgVal : sgVals.GetArray()) {
+    if (MolInterchange::stereoGrouplookup.find(sgVal["type"].GetString()) ==
+        MolInterchange::stereoGrouplookup.end()) {
+      throw FileParseException("Bad Format: bad stereogroup type");
+    }
+    const auto typ =
+        MolInterchange::stereoGrouplookup.at(sgVal["type"].GetString());
+    const auto &aids = sgVal["atoms"].GetArray();
+    std::vector<Atom *> atoms;
+    for (const auto &aid : aids) {
+      atoms.push_back(mol->getAtomWithIdx(aid.GetUint()));
+    }
+
+    if (!atoms.empty()) {
+      molSGs.emplace_back(typ, std::move(atoms));
+    }
+  }
+  mol->setStereoGroups(std::move(molSGs));
+}
+
+void readSubstanceGroups(RWMol *mol, const rj::Value &sgVals) {
+  PRECONDITION(mol, "no mol");
+
+  for (const auto &sgVal : sgVals.GetArray()) {
+    if (!sgVal.HasMember("properties") ||
+        !sgVal["properties"].HasMember("TYPE")) {
+      throw FileParseException(
+          "Bad Format: substance group does not have TYPE property");
+    }
+
+    auto sgType = sgVal["properties"]["TYPE"].GetString();
```

Comment:
Should we check whether the SGroup type is one of the known types, using `SubstanceGroupChecks::isValidType` ?

Also, after parsing the properties, we might want to check the subtype and "Connect" type, if these are present.